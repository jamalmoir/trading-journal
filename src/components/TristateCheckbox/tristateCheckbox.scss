@import "../../custom.scss";

.tristate-checkbox {
  width: 1em;
  height: 1em;
  border: 1px solid #ced4da;
  background: white;
  display: flex;
  justify-content: center;
  align-content: center;
  text-align: center;

  i {
    font-size: 0.75em;
    margin-top: 0.114em;
  }

  &.checked-false {
    background: $danger;

    i {
      color: $danger-light;
    }
  }

  &.checked-true {
    background: $success;

     i {
       color: $success-light
     }
  }
}