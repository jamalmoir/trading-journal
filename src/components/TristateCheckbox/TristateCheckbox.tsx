import * as React from 'react';
import { useState } from 'react';
import './tristateCheckbox.scss';


interface TristateCheckboxProps {
	className?: string;
	id?: string;
	onClick?: (val: boolean | null) => void;
	initial?: boolean | null;
}

const stateMap = new Map([[null, true], [true, false], [false, null]]);
const stateClassMap = new Map([[null, 'unchecked'], [true, 'checked-true'], [false, 'checked-false']]);


export const TristateCheckbox = (props: TristateCheckboxProps) => {
	const initialState = typeof props.initial !== 'undefined'  ? props.initial : null;
	const [checked, setChecked] = useState(initialState)
	const [checkedClass, setCheckedClass] = useState(stateClassMap.get(initialState));

	let handleClick = () => {
		let nextState = stateMap.get(checked);
		setChecked(nextState);
		setCheckedClass(stateClassMap.get(nextState));
		props.onClick(nextState);
	}

	return (
		<div id={ props.id }
				 className={ props.className + ' tristate-checkbox ' + checkedClass }
				 onClick={ handleClick }>
		</div>
	)
}