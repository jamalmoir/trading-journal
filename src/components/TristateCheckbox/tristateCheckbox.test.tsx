import * as React from 'react'
import { TristateCheckbox } from './TristateCheckbox'
import { shallow } from 'enzyme'

describe('<TristateCheckbox />', () => {
  it('matches snapshot', () => {
    const component = shallow(<TristateCheckbox />)
    expect(component).toMatchSnapshot()
  })
})
