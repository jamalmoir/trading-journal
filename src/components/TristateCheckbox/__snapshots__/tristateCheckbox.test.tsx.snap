// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TristateCheckbox /> matches snapshot 1`] = `
<div
  className="undefined tristate-checkbox unchecked"
  onClick={[Function]}
/>
`;
